assets/js/blocks/classic-template/index.tsx

```diff
@@ -99,51 +105,124 @@ const Edit = ( { clientId, attributes }: Props ) => {
 	);
 };
 
-/**
- * The 'WooCommerce Legacy Template' block was renamed to 'WooCommerce Classic Template', however, the internal block
- * name 'woocommerce/legacy-template' needs to remain the same. Otherwise, it would result in a corrupt block when
- * loaded for users who have customized templates using the legacy-template (since the internal block name is
- * stored in the database).
- *
- * See https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5861 for more context
- */
-registerBlockType( 'woocommerce/legacy-template', {
-	title: __( 'WooCommerce Classic Template', 'woo-gutenberg-products-block' ),
-	icon: (
-		<Icon icon={ box } className="wc-block-editor-components-block-icon" />
-	),
-	category: 'woocommerce',
-	apiVersion: 2,
-	keywords: [ __( 'WooCommerce', 'woo-gutenberg-products-block' ) ],
-	description: __(
-		'Renders classic WooCommerce PHP templates.',
-		'woo-gutenberg-products-block'
-	),
-	supports: {
-		align: [ 'wide', 'full' ],
-		html: false,
-		multiple: false,
-		reusable: false,
-		inserter: false,
-	},
-	example: {
-		attributes: {
-			isPreview: true,
+const templates = Object.keys( TEMPLATES );
+
+const registerClassicTemplateBlock = ( template?: string ) => {
+	registerBlockType( BLOCK_SLUG, {
+		title: template
+			? TEMPLATES[ template ].title
+			: __(
+					'WooCommerce Classic Template',
+					'woo-gutenberg-products-block'
+			  ),
+		icon: (
+			<Icon
+				icon={ box }
+				className="wc-block-editor-components-block-icon"
+			/>
+		),
+		category: 'woocommerce',
+		apiVersion: 2,
+		keywords: [ __( 'WooCommerce', 'woo-gutenberg-products-block' ) ],
+		description: __(
+			'Renders classic WooCommerce PHP templates.',
+			'woo-gutenberg-products-block'
+		),
+		supports: {
+			align: [ 'wide', 'full' ],
+			html: false,
+			multiple: false,
+			reusable: false,
+			inserter: true,
```

Comment:
This should be feature gated, otherwise, the block will be always available in the inserter when the block phase is less than 3.